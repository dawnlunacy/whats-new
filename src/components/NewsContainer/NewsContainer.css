.newsContainer {
    display: grid;
    grid-gap: 50px;
    grid-template-columns: repeat(2, 1fr);
    padding: 20px;
}