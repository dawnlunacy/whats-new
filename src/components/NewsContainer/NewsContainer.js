import React from 'react';
import './NewsContainer.css';
import NewsArticle from '../NewsArticle/NewsArticle';

const NewsContainer = (props) => {
    const news = props.news.map(story => {
        return <NewsArticle 
                    key={story.id}
                    img={story.img}
                    headline={story.headline}
                    description={story.description}
                    url={story.url}
                />
        })
    return (
        <section className="newsContainer"> 
            {news}
        </section>
       
        
    
 
    )
}
    


export default NewsContainer;