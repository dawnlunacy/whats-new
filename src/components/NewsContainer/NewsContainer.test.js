import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
    it('should match the news container snapshot with all data passed in correctly', () => {
        const articles = [
            {
            id: 11,
            headline: "PugSweaterCeption",
            img: "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSz_nzKAPAFYGHX6mSTun5HTDv3H84TxGOp9XLXovVMRQC1isQdUw",
            desciption: " Pug breaks the internet with cuteness",
            url: "https://imgur.com/t/pugly/rYptNJ9"
        },
        {
            id: 22,
            headline: " Robbie's Favorite",
            img: "https://i.pinimg.com/originals/df/ad/d7/dfadd7765de8e922206401f0529cae9a.jpg",
            desciption: "Pugs in Sweater Take Over The World",
            url: "https://www.pinterest.nz/pin/575546027375685533/"
        }]
        const wrapper = shallow(<NewsContainer 
            news = { articles }/>)

    expect(wrapper).toMatchSnapshot();

    })

})