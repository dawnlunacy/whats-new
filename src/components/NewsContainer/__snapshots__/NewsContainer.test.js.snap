// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the news container snapshot with all data passed in correctly 1`] = `
<section
  className="newsContainer"
>
  <NewsArticle
    headline="PugSweaterCeption"
    img="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSz_nzKAPAFYGHX6mSTun5HTDv3H84TxGOp9XLXovVMRQC1isQdUw"
    key="11"
    url="https://imgur.com/t/pugly/rYptNJ9"
  />
  <NewsArticle
    headline=" Robbie's Favorite"
    img="https://i.pinimg.com/originals/df/ad/d7/dfadd7765de8e922206401f0529cae9a.jpg"
    key="22"
    url="https://www.pinterest.nz/pin/575546027375685533/"
  />
</section>
`;
